src/fhirtypes/ElementDefinition.ts

```diff
@@ -846,6 +846,19 @@ export class ElementDefinition {
     return true;
   }
 
+  /**
+   * Checks if an element is fixed by a pattern[x] on its direct parent
+   * @returns {any} the value the element is fixed to by its parent, undefined if value is not fixed
+   */
+  fixedByParent(): any {
+    const parent = this.parent();
+    const patternKey = parent ? Object.keys(parent).find(k => k.startsWith('pattern')) : null;
+    if (patternKey) {
+      const patternValue: any = parent[patternKey as keyof ElementDefinition];
+      return patternValue[this.path.replace(`${parent.path}.`, '')];
+    }
+  }
```

Comment:
I noticed that this looks like it only goes up one parent to check.  Of course, the pattern could be several levels up (at a grandparent, for example).  So, does this only need to look up one level because other code (`validateValueAtPath`?) traverses the whole path -- so the grandparent values eventually make their way down to the child?

Reply:
When we set values on the instance from the StructureDefinition, the values are set based on paths that are used in the rules being used to create an instance. For each path that is contained in a rule, we traverse the whole path piece by piece. So if the path is `Patient.maritalStatus.coding.code`, we will set fixed values on the instance in the function `setFixedValuesFromStructureDefinition` using each part of that path, ie `Patient.maritalStatus`, then `Patient.maritalStatus.coding`, `Patient.maritalStatus.coding.code`. So yes if the pattern is on `Patient.maritalStatus` and that pattern fixes `Patient.maritalStatus.coding.code`, the `code` still gets fixed, even though the `fixedByParent` function only looks up one level.